src/libraries/Microsoft.PowerFx.Interpreter/Functions/LibraryTable.cs

```diff
@@ -462,6 +519,44 @@ private static bool IsValueTypeErrorOrBlank<T>(FormulaValue val)
             return val is T || val is BlankValue || val is ErrorValue;
         }
 
+        private static FormulaValue DistinctValueType<TPFxPrimitive, TDotNetPrimitive>(List<(DValue<RecordValue> row, FormulaValue sortValue)> pairs, IRContext irContext)
+            where TPFxPrimitive : PrimitiveValue<TDotNetPrimitive>
+            where TDotNetPrimitive : IComparable<TDotNetPrimitive>
+        {
+            var values = new Dictionary<TDotNetPrimitive, DValue<RecordValue>>();
+            var name = ((TableType)irContext.ResultType).SingleColumnFieldName;
+            var type = ((TableType)irContext.ResultType).SingleColumnFieldType;
+
+            // Have to handle null
+            var addBlankValue = false;
+
+            foreach (var (row, distinctValue) in pairs)
+            {
+                if (distinctValue is BlankValue)
+                {
+                    addBlankValue = true;
+                    continue;
+                }
+
+                var key = (TDotNetPrimitive)distinctValue?.ToObject();
```

Comment:
We probably need some platform functionality for this. 
It's important that this is limited to just primitives (ToObject() on records is not reliable).